addons/controllers/addon_controller.go

```diff
@@ -235,6 +235,15 @@ func (r *AddonReconciler) reconcileDelete(
 		return ctrl.Result{}, kerrors.NewAggregate(errors)
 	}
 
+	_, isManagmentCluster := cluster.Labels[constants.ManagementClusterRoleLabel]
+	if !isManagmentCluster {
```

Comment:
Is it possible to add to some test in unit test to check for existence of finalizer and removal of finalizer ?

Reply:
Done